File: backend-api/src/models/Restaurant.js
const mongoose = require('mongoose');

const MenuItemSchema = new mongoose.Schema({
  name: { type: String, required: true },
  price: { type: Number, required: true },
  description: { type: String, default: '' },
  imageUrl: { type: String, default: '' }
});

const RestaurantSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  imageUrl: { type: String, required: true },
  rating: { type: Number, default: 0.0 },
  reviewCount: { type: Number, default: 0 },
  category: { type: String, required: true },
  description: { type: String, default: '' },
  menuItems: [MenuItemSchema]
}, {
  timestamps: true
});

module.exports = mongoose.model('Restaurant', RestaurantSchema);

File: app/src/main/java/com/example/foodboxd/domain/model/Restaurant.kt
package com.example.foodboxd.domain.model

data class Restaurant(
    val id: String,
    val name: String,
    val imageUrl: String,
    val rating: Double,
    val reviewCount: Int,
    val category: String,
    val description: String,
    val isFavorite: Boolean,
    val menuItems: List<MenuItem> = emptyList()
)

data class MenuItem(
    val id: String,
    val name: String,
    val price: Double,
    val description: String,
    val imageUrl: String
)
